File: src/main/java/org/skyhype/core/SkyhypePlugin.java
package org.skyhype.core;

import org.skyhype.core.manager.CommandManager;
import org.skyhype.core.manager.DataManager;
import org.skyhype.core.manager.ItemManager;
import xyz.oribuin.orilibrary.OriPlugin;

public class SkyhypePlugin extends OriPlugin {
    private static SkyhypePlugin instance;

    @Override
    public void enablePlugin() {
        instance = this;

        getServer().getScheduler().runTaskAsynchronously(this, () -> {
            this.getManager(CommandManager.class);
            this.getManager(DataManager.class);
            this.getManager(ItemManager.class);
        });

    }

    @Override
    public void disablePlugin() {

    }

    public static SkyhypePlugin getInstance() {
        return instance;
    }
}

File: src/main/java/org/skyhype/core/SkyhypePlugin.kt
package org.skyhype.core

import xyz.oribuin.orilibrary.OriPlugin
import org.skyhype.core.SkyhypePlugin
import java.lang.Runnable
import org.skyhype.core.manager.DataManager
import org.skyhype.core.manager.ItemManager
import org.skyhype.core.listener.JoinEvents
import org.skyhype.core.manager.CommandManager

class SkyhypePlugin : OriPlugin() {
    override fun enablePlugin() {

        server.scheduler.runTaskAsynchronously(this, Runnable {
            getManager(CommandManager::class.java)
            getManager(DataManager::class.java)
            getManager(ItemManager::class.java)
        })

        // Register Listeners
        JoinEvents(this)
    }

    override fun disablePlugin() {}
}